wevote/WebApp
# [WV-312] replace placeholder avatar with candidate initials [TEAM REVIEW]

## Summary

This PR introduces a feature that generates avatars for supporters based on their names when no profile image is available. The avatars display the initials of the supporter’s name and use a color derived from the name to ensure visual distinction.

## Description

- Implemented a `speakerDisplayNameToInitials` function that generates initials and background colors for avatars based on the `supporter_name`.
- Updated the `CommentWrapper` component to use the `speakerDisplayNameToInitials` function when no profile image is available.
- Refactored the code to ensure proper handling of single-word and multi-word supporter names.
- Added utility functions `speakerDisplayNameToAvatarColor` and `speakerDisplayNameToInitials` to centralize the avatar creation logic.

## Testing Instructions

1. Run the application and navigate to the campaign page with supporters.
2. Verify that supporters without profile images have avatars with initials and unique background colors.
3. Test with both single-word and multi-word supporter names to ensure initials are correctly generated.

## Screenshots / GIFs

![image](https://github.com/user-attachments/assets/bb49c1f1-117a-479d-b170-7d1fc2d81c75)
